keyboards/gmmk/pro/rev1/ansi/keymaps/thystips/keymap.c

```diff
@@ -0,0 +1,202 @@
+    ),
+
+   [_RAISE] = LAYOUT(
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,          _______,
+        _______,          _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______, _______, _______,
+        _______, _______, _______,                            _______,                            _______, _______, _______, _______, _______, _______
+    ),
+};
+
+#if defined(ENCODER_ENABLE) && !defined(ENCODER_DEFAULTACTIONS_ENABLE) // Encoder Functionality when not using userspace defaults
+void encoder_action_rgbhue(bool clockwise) {
+    if (clockwise)
+        rgblight_increase_hue_noeeprom();
+    else
+        rgblight_decrease_hue_noeeprom();
+}
+
+__attribute__((weak)) bool encoder_update_keymap(uint8_t index, bool clockwise) {
+    return true;
+}
+
+bool encoder_update_user(uint8_t index, bool clockwise) {
+    if (!encoder_update_keymap(index, clockwise)) {
+        return false;
+    }
+    if (index != ENCODER_DEFAULTACTIONS_INDEX) {
+        return true;
+    } // exit if the index doesn't match
+    uint8_t mods_state = get_mods();
+    if (mods_state & MOD_BIT(KC_LSFT)) { // If you are holding L shift, encoder changes layers
+        encoder_action_layerchange(clockwise);
+    } else if (mods_state & MOD_BIT(KC_RSFT)) { // If you are holding R shift, Page up/dn
+        unregister_mods(MOD_BIT(KC_RSFT));
+        encoder_action_navpage(clockwise);
+        register_mods(MOD_BIT(KC_RSFT));
+    } else if (mods_state & MOD_BIT(KC_LCTL)) { // if holding Left Ctrl, navigate next/prev word
+        encoder_action_navword(clockwise);
+    } else if (mods_state & MOD_BIT(KC_RCTL)) { // if holding Right Ctrl, change rgb hue/colour
+        encoder_action_rgbhue(clockwise);
+    } else if (mods_state & MOD_BIT(KC_LALT)) { // if holding Left Alt, change media next/prev track
+        encoder_action_mediatrack(clockwise);
+    } else {
+        switch (get_highest_layer(layer_state)) {
+            case _FN1:
+#    ifdef IDLE_TIMEOUT_ENABLE
+                timeout_update_threshold(clockwise);
+#    endif
+                break;
+            default:
+                encoder_action_volume(clockwise); // Otherwise it just changes volume
+                break;
+        }
+    }
+    return false;
+}
+#endif // ENCODER_ENABLE && !ENCODER_DEFAULTACTIONS_ENABLE
```

Comment:
This is the same callback as what is in your userspace (and most of the same code).  Pretty sure this will error during compiling, too.
```suggestion
```